keras_cv/utils/bbox_test.py

```diff
@@ -74,3 +74,24 @@ def test_xywh_to_corner(self):
         self.assertAllClose(
             bbox.xywh_to_corners(padded_xywh_bbox_3d), padded_corner_bbox_3d
         )
+
+    def test_bbox_padding(self):
+        bboxes = [[1, 2, 3, 4], [5, 6, 7, 8]]
+        target_shape = [3, 4]
+        result = bbox.pad_bbox_batch_to_shape(bboxes, target_shape)
+        self.assertAllClose(result, [[1, 2, 3, 4], [5, 6, 7, 8], [-1, -1, -1, -1]])
+
+        target_shape = [2, 5]
+        result = bbox.pad_bbox_batch_to_shape(bboxes, target_shape)
+        self.assertAllClose(result, [[1, 2, 3, 4, -1], [5, 6, 7, 8, -1]])
+
+        # Make sure to raise error if the rank is different between bbox and target
+        # shape
+        with self.assertRaisesRegexp(
+                ValueError, 'Target shape should have same rank'):
+            bbox.pad_bbox_batch_to_shape(bboxes, [1, 2, 3])
+
+        # Make sure to not pad the dim if the target shape is smaller
+        target_shape = [3, 2]
+        result = bbox.pad_bbox_batch_to_shape(bboxes, target_shape)
+        self.assertAllClose(result, [[1, 2, 3, 4], [5, 6, 7, 8], [-1, -1, -1, -1]])
```

Comment:
Is this the expected result `[[1, 2, 3, 4], [5, 6, 7, 8], [-1, -1, -1, -1]]`?

Reply:
yes, the original shape is [2, 4] and since we couldn't shrink the second dim [4], the result shape is [3, 4].

Reply:
what was your expected result in this case? I could raise an error if the target shape has smaller dim when comparing to input bboxs.

Reply:
Yes probably it could be useful I think someone could go crazy when he is debugging a large code base. 
Do we need to support negative pad?

Reply:
E.g. `tf.pad` has 
`InvalidArgumentError: must be non-negative` for negative inputs

Reply:
Ack. Let me update the code/docstring/unittest

Reply:
Done.